From 19e7c78b71b2f7a73f2d6d86592dc560dcd245cc Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sat, 15 Dec 2012 21:55:50 +0800
Subject: libstagefright: enhance error handle for NuCachedSource2

BZ: 74681

enhance error handle for NuCachedSource2.When invalid offset(<0) resulted from a asf clip without
simple index object is passed to readAt,return error.

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I38a13a62bc35683c099d14e0c27dbe66f2684ab5
Orig-Change-Id: I400e045bb983f1fcbd169c6b4b8b925ad486fa10
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/NuCachedSource2.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/NuCachedSource2.cpp b/media/libstagefright/NuCachedSource2.cpp
index 05e599b..81c9a92 100644
--- a/media/libstagefright/NuCachedSource2.cpp
+++ b/media/libstagefright/NuCachedSource2.cpp
@@ -468,6 +468,10 @@ ssize_t NuCachedSource2::readAt(off64_t offset, void *data, size_t size) {
 
     Mutex::Autolock autoLock(mLock);
 
+    if (offset < 0) {
+        return ERROR_OUT_OF_RANGE;
+    }
+
     // If the request can be completely satisfied from the cache, do so.
 
     if (offset >= mCacheOffset
-- 
1.7.4.1

